AppCenterData/AppCenterData/MSData.m

```diff
@@ -915,6 +923,14 @@ - (void)synchronizeLocalCacheWithCosmosDbWithToken:(MSTokenResult *)token
   if (shouldDeleteLocalCache) {
     [self.dataOperationProxy.documentStore deleteWithToken:token documentId:documentId];
   }
+
+  // If the Remote operation is set
+  @synchronized(self) {
+    id<MSRemoteOperationDelegate> strongDelegate = self.remoteOperationDelegate;
+    if (strongDelegate) {
+      [strongDelegate data:self didCompletePendingOperation:pendingOperation forDocument:document withError:documentWrapper.error];
```

Comment:
As a good practice the execution of the delegate is happening outside of the `synchronized`. This is because what the delegate is doing is either unknown or uneasy to track. In this case it's unknown since it's developer code.
This means lines 930 to 932 should be moved after the `synchronized` block.

Reply:
That will also help in keeping the `synchronized` block to it's minimal.

Reply:
Yeah.. This is caused by my comment over offline communication channel.
My intention was to ask to change this as below.
```
  // If the Remote operation is set
  id<MSRemoteOperationDelegate> strongDelegate;
  @synchronized(self) {
    strongDelegate = self.remoteOperationDelegate;
  }
  if (strongDelegate) {
    [strongDelegate data:self didCompletePendingOperation:pendingOperation forDocument:document withError:documentWrapper.error];
  }
```